### ISSUE TITLE
Animation types have errors when animating CSS custom properties

### ISSUE THREAD

**Issue by spmonahan** [microsoft/griffel#417]
When animating CSS custom properties Typescript raises an error:

> Type '{ "--my-var": string; }' is not assignable to type 'GriffelResetStylesCSSProperties'.
  Object literal may only specify known properties, and '"--my-var"' does not exist in type 'GriffelResetStylesCSSProperties'.ts(2322)

This issue affects both `makeStyles` and `makeResetStyles`. I believe it originates from #413 as this code worked prior to the type change.

Repro: https://codesandbox.io/s/griffel-css-animation-myhrq4?file=/src/index.tsx

NOTE: repro will only run in a Chrome and Edge at the time of this writing (and [maybe Safari](https://ishoudinireadyyet.com/), but haven't tested there).

This code is taking advantage of the fact that [registered custom properties are animatable](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_animated_properties#animating_custom_properties). Technically custom properties can be used in animations without registering them, but animations are discrete in that case.

Would be nice for Griffel types to support custom properties in animations.

**Comment by layershifter** [comment]
> I believe it originates from #413 as this code worked prior to the type change.

Yeah, the source of the problem seems to be true. @spmonahan do you want to work on the issue resolution?

**Comment by spmonahan** [comment]
Happy to help but I won't have time for a couple weeks. It's not blocking me at the moment since I can cast the problem away (not ideal but it's fine for now). Please go ahead and assign it to me and I'll get it on my backlog.
